components/marketplace/buy-fractional-order-form.tsx

```diff
@@ -83,9 +84,10 @@ export const BuyFractionalOrderForm = ({
   const getUnitsToBuy = (percentageAmount: string) => {
     try {
       const hypercertUnits = BigInt(hypercert.units || 0);
-      const percentageAsBigInt = BigInt(Number(percentageAmount) * 100);
+      const percentageAsBigInt = parseUnits(percentageAmount, 8);
```

Comment:
The 8 is a magic number. I think it would be better to put that into a constant either here in the form or alongside `DEFAULT_NUM_FRACTIONS`

Reply:
How do you feel about `percentageInWei`?

Reply:
> It's not Wei though, right? As we're using `10^8` and not `10^18`.

You're right, my bad!